python-sdk/src/astro/databases/mssql.py

```diff
@@ -0,0 +1,359 @@
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING
+
+import pandas as pd
+import sqlalchemy
+from airflow.providers.microsoft.mssql.hooks.mssql import MsSqlHook
+from sqlalchemy.sql import ClauseElement
+
+from astro.constants import DEFAULT_CHUNK_SIZE, LoadExistStrategy, MergeConflictStrategy
+from astro.databases.base import BaseDatabase
+from astro.settings import MSSQL_SCHEMA
+from astro.table import BaseTable, Metadata
+
+DEFAULT_CONN_ID = MsSqlHook.default_conn_name
+if TYPE_CHECKING:  # pragma: no cover
+    from sqlalchemy.engine.cursor import CursorResult
+
+
+class MssqlDatabase(BaseDatabase):
+    DEFAULT_SCHEMA = MSSQL_SCHEMA
+
+    _create_schema_statement: str = """
+    IF NOT EXISTS (SELECT 1 FROM sys.schemas WHERE name = '{0}')
+    BEGIN
+        EXEC( 'CREATE SCHEMA {0}' );
+    END
+    """
+    _truncate_table_statement: str = "TRUNCATE TABLE {0}"
+
+    # CTAS IS NOT SUPPORTED IN SQL SERVER. HENCE USE SELECT * INTO
+    # SELECT * INTO FROM A SUBQUERY WILL WORK IN SQL SERVER ONLY WHEN AN ALIAS IS ADDED
+    _create_table_statement: str = """
+    IF (NOT EXISTS (SELECT *
+                 FROM INFORMATION_SCHEMA.TABLES
+                 WHERE TABLE_SCHEMA = '{0}'
+                 AND   TABLE_NAME    = '{1}'
+                 AND   TABLE_CATALOG = '{2}'))
+    BEGIN
+        SELECT * INTO  {2}.{0}.{1} FROM ({3}) as astro;
+    END
+    """
+
+    _queries_requiring_autocommit: list = ["alter", "select * into"]
```

Comment:
@phanikumv Why do we need this only for these queries?

Reply:
because alter , select * into, drop table are considered as DDL(they change the system catalog). The pymssql's sqlalchemy considers autocommit as False by default which causes these DDL's to not execute

Reply:
Added comment explaining this in [88008cd](https://github.com/astronomer/astro-sdk/pull/1538/commits/88008cd4593127db9111768fa2dbef51502d955d)